### PR TITLE
Setup 1% AB test of Updated Xaxis bidder

### PR DESCRIPTION
**Pull request by ioanna0**
## What does this change?
Reactivate the Xaxis prebid adaptor with updated IDs so we can remove them from Ozone

Add it behind 1% AB test as we want to ensure that adding this bidder does not impact the performance of prebid (D&I will be working on the analysis)

Apply a 5% bid boost

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/bid-config.spec.js [hunk 1]
```diff
@@ -524,3 +525,79 @@ describe('triplelift adapter', () => {
...
+
+    test('should return -1 if no cases match', () => {
+        expect(getImprovePlacementId([[1, 2]])).toBe(-1);
+    });
+
+    test('should return the expected values for desktop device', () => {
+        getBreakpointKey.mockReturnValue('D');
+        containsMpuOrDmpu.mockReturnValueOnce(true);
+        containsMpuOrDmpu.mockReturnValueOnce(true);
+        containsMpuOrDmpu.mockReturnValue(false);
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/header-bidding/prebid/bid-config.spec.js:558]: These mocks can be [chained][], and maybe extracted to the `beforeEach`?

```suggestion
        containsMpuOrDmpu
            .mockReturnValueOnce(true);
            .mockReturnValueOnce(true);
            .mockReturnValue(false);
```

[chained]: https://jestjs.io/docs/en/mock-function-api#mockfnmockreturnvalueoncevalue

File: static/src/javascripts/projects/common/modules/experiments/tests/updated-xaxis-prebid.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// @flow strict
```
Comment (mxdvl) [inline static/src/javascripts/projects/common/modules/experiments/tests/updated-xaxis-prebid.js:1]: I thought we removed `flow`?

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js [hunk 3]
```diff
@@ -93,6 +93,9 @@ const initialise = (window) => {
                     },
                 },
             ],
+            bidCpmAdjustment : (bidCpm) => {
+                return bidCpm * 0.5;
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js:97]: If I understand [the docs][] correctly, it should be 105% for a 5% boost:

```suggestion
                return bidCpm * 1.05;
```

[the docs]: https://docs.prebid.org/dev-docs/publisher-api-reference.html#22-bidcpmadjustment

**Review by mxdvl — CHANGES_REQUESTED** [review]
I think we need to confirm the logic around the bid adjustment!

See https://github.com/guardian/frontend/pull/23533#discussion_r572259049

File: static/src/javascripts/projects/common/modules/experiments/tests/updated-xaxis-prebid.js (hunk 2 above)
Comment (ioanna0) [reply static/src/javascripts/projects/common/modules/experiments/tests/updated-xaxis-prebid.js]: oops!

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js (hunk 3 above)
Comment (ioanna0) [reply static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js]: good catch, have changed it to 1.05

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/bid-config.spec.js [hunk 4]
```diff
@@ -524,3 +525,69 @@ describe('triplelift adapter', () => {
...
+
+    const generateTestIds = () => {
+        const prebidSizes = [
+            [[300, 250]],
+            [[300, 600]],
+            [[970, 250]],
+            [[728, 90]],
+            [[1, 2]],
+        ];
+        return prebidSizes.map(getXaxisPlacementId);
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/header-bidding/prebid/bid-config.spec.js:554]: There’s been some chat around this recently, and the danger around the fact that map has arity 3: https://jakearchibald.com/2021/function-callback-risks/

**Review by mxdvl — APPROVED** [review]
Looks good!

**Pull request merged**